src/wp-includes/functions.php

```diff
@@ -2920,18 +2925,18 @@ function wp_upload_bits( $name, $deprecated, $bits, $time = null ) {
 		)
 	);
 	if ( ! is_array( $upload_bits_error ) ) {
-		$upload['error'] = $upload_bits_error;
-		return $upload;
+		$upload_dir['error'] = $upload_bits_error;
+		return $upload_dir;
 	}
 
-	$filename = wp_unique_filename( $upload['path'], $name );
+	$filename = wp_unique_filename( $upload_dir['path'], $name );
 
-	$new_file = $upload['path'] . "/$filename";
+	$new_file = $upload_dir['path'] . "/$filename";
 	if ( ! wp_mkdir_p( dirname( $new_file ) ) ) {
```

Comment:
This function attempts to create the directory but it doesn't appear that `_wp_handle_upload()` does (correct me if I am wrong).

Given these changes allow developers to define the uploads dir, I think it should either:

* attempt to create a directory similar to the sideloading function
* add a new error message if the directory doesn't exist, it can use the  same string as in the other functions.

If the second approach is taken then the override docs will need to include a note that the directory needs to be created before attempting an upload.

Reply:
> I'm a bit unsure which fix would be better here.

Same to be 100% honest, I could go either way too. If the upload handlers attempt to create the directory then it seems redundant that `wp_upload|font_dir()` does too, but this change would complicate that.

Let's sleep on it and hopefully our unconscious can decide for us overnight.

Reply:
> Let's sleep on it and hopefully our unconscious can decide

Great idea! Unfortunately my unconsciousness did not take that opportunity :) 

Joking aside, I tend to very slightly prefer letting plugins handle creation and confirmation of the uploads directory. That would give them a little more freedom in how to do it and whether and how to cache it. Of course they can just use `wp_mkdir_p()`. Thinking that checking the type and with `is_dir()`, and adding a longer description there would be sufficient.

Also tried to figure out if there are any differences between PHPs `mkdir()` and `is_dir()` when using handlers. Think there may have been in the past but seems now there isn't (or at least I wasn't able to find anything).